bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1239,142 @@ mod test {
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_no_checksig_returns_none() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let no_checksig = Script::builder().push_key(&key).into_script();
+        assert_eq!(no_checksig.p2pk_pubkey_bytes(), None);
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_emptry_script_returns_none() {
+        let empty_script = Script::builder().into_script();
+        assert!(empty_script.p2pk_pubkey_bytes().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_no_key_returns_none() {
+        // scripts with no key should return None
+        let no_push_bytes = Script::builder().push_opcode(OP_CHECKSIG).into_script();
+        assert!(no_push_bytes.p2pk_pubkey_bytes().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_different_op_code_returns_none() {
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let different_op_code = Script::builder().push_key(&key).push_opcode(OP_NOP).into_script();
+        assert!(different_op_code.p2pk_pubkey_bytes().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_incorrect_key_size_returns_none() {
+        // 63 byte key
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1";
+        let invalid_p2pk_script = Script::builder()
+            .push_slice(malformed_key.as_bytes())
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+        assert!(invalid_p2pk_script.p2pk_pubkey_bytes().is_none());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_invalid_key_returns_some() {
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1ux";
+        let invalid_key_script = Script::builder()
+            .push_slice(malformed_key.as_bytes())
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+        assert!(invalid_key_script.p2pk_pubkey_bytes().is_some());
+    }
+
+    #[test]
+    fn test_p2pk_pubkey_bytes_compressed_key_returns_expected_key() {
+        let compressed_key_str = "0311db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5c";
+        let key = PublicKey::from_str(compressed_key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+    }
+
+    #[test]
+    fn test_p2pk_public_key_valid_key_and_valid_script_returns_expected_key() {
```

Comment:
This is awesome, you took on board what reviewers said and improved the PR, respect.

Here you added `test_` to the front of the test name. This prefixing is done in a bunch of places in our codebase but it is not correct Rust convention. We can argue all day about whether it should be convention or not but I'm of the opinion that if a language has conventions like this its easier to just follow them. In case you are want to know why the convention, its so that the tests don't stutter (since by convention we keep unit tests in a module called `test`). Compare output of test run:
```
test blockdata::script::test::test_p2pk_pubkey_bytes_compressed_key_returns_expected_key ... ok
```
to
```
test blockdata::script::test::p2pk_pubkey_bytes_compressed_key_returns_expected_key ... ok
```

Its a minor thing but one of my pet-peeves. This message is intended to be read by other devs/maintainers too. I would appreciate a nomination for this weeks most anal github post :)

Reply:
Oh, this was a habit from how I learned testing. I'll change this up and make sure to follow it in the future